### PR TITLE
Paragraph extraction

### PR DESCRIPTION
**Pull request by quarterto**
some mildly unrelated stuff to increase type coverage

### CODE REVIEW

File: src/model/item.js [hunk 1]
```diff
@@ -20,7 +21,17 @@ module.exports = class Item extends ResultMapper {
 	}
 
 	get abstract(): ?string {
-		return this._data.openingHTML;
+		if(this._data.openingHTML) {
+			return this._data.openingHTML;
+		}
+
+		if(this._data.bodyHTML) {
+			const $ = cheerio.load(this._data.bodyHTML);
```
Comment (rowanbeentje) [inline src/model/item.js]: So *this* may cause a significant load increase compared to anything happening previously, particularly with a bunch of posts being loaded by a ton of users.  We have a tiny cache so at least it'll be rate-limited, let's see how it does :)

File: src/model/item.js [hunk 2]
```diff
@@ -20,7 +21,17 @@ module.exports = class Item extends ResultMapper {
...
+
+			const firstPara = $('p').first();
```
Comment (rowanbeentje) [inline src/model/item.js]: Might want to deal with no-first-para responses, because fastft content is Weird™

Comment (rowanbeentje) [reply src/model/item.js]: Probably all fine :)

File: src/model/item.js (hunk 1 above)
Comment (quarterto) [reply src/model/item.js]: okay so this is an enormous load increase. as in going from 30ms median, 70ms max to crashing `ab`. hmph

Comment (rowanbeentje) [reply src/model/item.js]: wot's the cache?

**Pull request merged**